### PR TITLE
Add handling for notifications due to invalid sub

### PR DESCRIPTION
**Pull request by karlenDimla**
Task/Issue URL: https://app.asana.com/0/0/1206170093899189/f

### Description
See attached task description

### Steps to test this PR

https://app.asana.com/0/0/1206184871859535/f

### CODE REVIEW

**Comment by karlenDimla** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #4000** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4000?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #4003** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4003?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #4005** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4005?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/4005?utm_source=stack-comment).

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/configuration/WgServerApi.kt [hunk 1]
```diff
@@ -44,43 +45,54 @@ class RealWgServerApi @Inject constructor(
...
-                } ?: false
-            }
-
-        val userPreferredLocation = netNetpEgressServersProvider.updateServerLocationsAndReturnPreferred()
-        val registerKeyBody = if (selectedServer != null) {
-            RegisterKeyBody(publicKey = publicKey, server = selectedServer)
-        } else if (userPreferredLocation != null) {
-            if (userPreferredLocation.cityName != null) {
-                RegisterKeyBody(publicKey = publicKey, country = userPreferredLocation.countryCode, city = userPreferredLocation.cityName)
+        val registerKeyBody = try {
```
Comment (karlenDimla) [inline network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/configuration/WgServerApi.kt:52]: This is one way of doing it. Another way would be to handle every 403 from every endpoint being called here (servers and locations). I figured this is the simpler approach

File: network-protection/network-protection-subscription-internal/src/main/java/com/duckduckgo/networkprotection/subscription/notification/SubsNetpDisabledNotificationBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import com.duckduckgo.networkprotection.impl.store.NetworkProtectionRepository
+import com.squareup.anvil.annotations.ContributesBinding
+import com.squareup.anvil.annotations.ContributesBinding.Priority.HIGHEST
+import javax.inject.Inject
+
+@ContributesBinding(
+    scope = VpnScope::class,
+    priority = HIGHEST,
+)
+class SubsNetpDisabledNotificationBuilder @Inject constructor(
```
Comment (karlenDimla) [inline network-protection/network-protection-subscription-internal/src/main/java/com/duckduckgo/networkprotection/subscription/notification/SubsNetpDisabledNotificationBuilder.kt:33]: Another way to do this is to let https://github.com/duckduckgo/Android/blob/9536f6666eb1644acc639265698bcd10abaa1929/network-protection/network-protection-subscription-internal/src/main/java/com/duckduckgo/networkprotection/subscription/notification/NetAccessRevokedNotificationScheduler.kt handle the logic for when the revoked access notif could be shown. However, we would also need to modify `NetPDisabledNotificationScheduler` to not show the disabled notif.

I prefer the option to isolate all this logic in the subs internal land for now.

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/configuration/WgServerApi.kt [hunk 3]
```diff
@@ -44,43 +45,54 @@ class RealWgServerApi @Inject constructor(
...
-                logcat { "Register key in $selectedServer" }
-                logcat { "Register key returned ${this.map { it.server.name }}" }
-                val server = this.firstOrNull()?.toWgServerData()
-                logcat { "Selected Egress server is $server" }
-                server
+                if (isSuccessful) {
+                    logcat { "Register key returned ${this.body()?.map { it.server.name }}" }
+                    val server = this.body()?.firstOrNull()?.toWgServerData()
+                    logcat { "Selected Egress server is $server" }
+                    netpRepository.disabledDueToAccessRevoked = false
```
Comment (aitorvs) [inline network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/configuration/WgServerApi.kt:90]: As discussed, we can try to move the `403` check into the `NetpWaitlistRequestInterceptor` and leave this class as is

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/store/NetworkProtectionRepository.kt [hunk 4]
```diff
@@ -36,6 +36,7 @@ interface NetworkProtectionRepository {
     var enabledTimeInMillis: Long
     var serverDetails: ServerDetails?
     var clientInterface: ClientInterface?
+    var disabledDueToAccessRevoked: Boolean
```
Comment (aitorvs) [inline network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/store/NetworkProtectionRepository.kt:39]: As discussed, maybe better to rename to `vpnAccessRevoked` as it VPN enable/disable doesn't related to whether the VPN entitlement is revoked or not (eg. VPN can be ON even if access is revoked)

File: network-protection/network-protection-subscription-internal/src/main/java/com/duckduckgo/networkprotection/subscription/notification/NetAccessRevokedNotificationScheduler.kt [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    override fun onVpnStarted(coroutineScope: CoroutineScope) {}
+
+    override fun onVpnStopped(
+        coroutineScope: CoroutineScope,
+        vpnStopReason: VpnStopReason,
+    ) {
+    }
+
+    override fun onVpnStartFailed(coroutineScope: CoroutineScope) {
+        super.onVpnStartFailed(coroutineScope)
```
Comment (aitorvs) [inline network-protection/network-protection-subscription-internal/src/main/java/com/duckduckgo/networkprotection/subscription/notification/NetAccessRevokedNotificationScheduler.kt:50]: Remove?
```suggestion
```

File: network-protection/network-protection-subscription-internal/src/main/java/com/duckduckgo/networkprotection/subscription/notification/NetpAccessRevokedNotiification.kt
Comment (aitorvs) [inline network-protection/network-protection-subscription-internal/src/main/java/com/duckduckgo/networkprotection/subscription/notification/NetpAccessRevokedNotiification.kt:1]: Typo in the filename `NetpAccessRevokedNotiification.kt` -> `NetpAccessRevokedNotification.kt`

**Review by aitorvs — APPROVED** [review]

**Pull request merged**